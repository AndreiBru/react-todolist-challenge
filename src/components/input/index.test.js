import React from 'react';
import renderer from 'react-test-renderer';
import { shallow } from 'enzyme';

import Input from './';

describe('Input', () => {
  it('renders and matches snapshot', () => {
    const comp = renderer.create(
      <Input
        name="email"
        value="name@gmail.com"
        placeholder="Your email address"
        className="one two"
      />,
    );

    expect(comp.toJSON()).toMatchSnapshot();
  });

  it('calls the onChange handler with the correct value', () => {
    const onChangeSpy = jest.fn();
    const comp = shallow(
      <Input
        name="email"
        value="name@gmail.com"
        placeholder="Your email address"
        className="one two"
        onChange={onChangeSpy}
      />,
    );

    const newValue = 'hey';

    expect(onChangeSpy).not.toHaveBeenCalled();

    comp.find('input').simulate('change', { target: { value: newValue } });

    expect(onChangeSpy).toHaveBeenCalledTimes(1);
    expect(onChangeSpy).toHaveBeenCalledWith(newValue);
  });
});
