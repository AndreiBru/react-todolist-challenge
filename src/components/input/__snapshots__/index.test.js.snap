// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Input renders and matches snapshot 1`] = `
<input
  className="one two"
  name="email"
  onChange={[Function]}
  placeholder="Your email address"
  type="text"
  value="name@gmail.com"
/>
`;
